tests/bats/tests.bats

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bats
+
+setup() {
+  cd ${BATS_TEST_DIRNAME}
+
+  if [ -x "$(command -v timeout)" ]; then
+    TIMEOUT=timeout
+  elif [ -x "$(command -v gtimeout)" ]; then
+    TIMEOUT=gtimeout
+  else
+    skip "timeout(gtimeout) is required"
+  fi
+
+  case "${BATS_TEST_NAME}" in
+  *hvt)
+    [ "${STACK}" != "hvt" ] && skip "hvt not built"
+    ;;
+  *spt)
+    [ "${STACK}" != "spt" ] && skip "spt not built"
+    ;;
+  *hw)
+    [ "${STACK}" != "qemu" ] && skip "hw not built"
+    ;;
```

Comment:
set defaults to skip incase of typo check

Reply:
done